ntoskrnl/kd64/kdprint.c

```diff
@@ -343,31 +343,22 @@ KdpPrint(
     STRING OutputString;
     CHAR CapturedString[512];
 
-    /* Assume failure */
-    *Handled = FALSE;
-
-#if (NTDDI_VERSION >= NTDDI_VISTA)
-    if ((ComponentId >= KdComponentTableSize) && (ComponentId < MAXULONG))
+    /* For such case, we just check the mask against system-wide one */
+    if (ComponentId >= KdComponentTableSize)
     {
-        /* Use the default component ID */
-        Mask = &Kd_DEFAULT_Mask;
-        // Level = DPFLTR_INFO_LEVEL; // Override the Level.
+        ComponentId = MAXULONG;
     }
-#endif
-    /* Convert Level to bit field if required */
-    if (Level < 32) Level = 1 << Level;
-    Level &= ~DPFLTR_MASK;
-
-    /* Validate the mask */
-    if (!(Kd_WIN2000_Mask & Level) ||
```

Comment:
Ok the actual one-liner patch was:
```diff
-    if (!(Kd_WIN2000_Mask & Level) ||
+    if (!(Kd_WIN2000_Mask & Level) &&
```

Reply:
Yep, which is equal to what `NtQueryDebugFilterState` does. That happened because the same logic is spread across two functions :)